### DISCUSSION TITLE
Embed stream in webpage oEmbed

### DISCUSSION THREAD

**Discussion by natuk** [gerbera/gerbera#2629] [Q&A] [answered]
Seeking some advice on best practices for embedding streams from Gerbera into a webpage, i.e. I wish to create a webpage with some content and embed a video hosted in Gerbera. What would be a good way of doing it? Some transcoding to save bandwidth might be necessary. Related: the [oEmbed](https://oembed.com/) standard provides an API for producing the embedding code, but that is for a later step.

Many thanks for your time.

**Comment by KarlStraussberger** [comment, marked as answer]
Sounds like a weird idea.

I'd recommend to hide everything behind a reverse proxy (https://docs.gerbera.io/en/stable/run.html#reverse-proxy-setup).

If you want to transcode you need to set up the profile (https://docs.gerbera.io/en/stable/transcoding.html, https://docs.gerbera.io/en/stable/config-transcode.html). In order to run it from your web site you have to add `/tr/profile` in the uri you embed.

**Reply by natuk** [reply]
Thank you for this.

The use case is, for example, publishing a text narrative documenting an event on a webpage, and including several short videos shot at the event. These videos are not hosted with the website, but need to be streamed from a media server.

I think I can imagine how it could work, but I am not clear on:

* how the stream URI is provided/constructed by Gerbera
* what kind of authentication would be applied on that stream

These may become obvious once I set it up, just doing some feasibility research. Thanks again.

**Reply by KarlStraussberger** [reply]
The uri looks like the `http://server:port/content/media/object_id/1147284/res_id/0/group/web/ext/file.mpeg`

The object id is an automatic database key. It is constant as long as you do not change the file.

You should use a reverse proxy to allow access only to well known files.

There is no authentication at all which can be used in your use case.
